### DISCUSSION TITLE
Configuration methods cannot accept parameters

### DISCUSSION THREAD

**Discussion by cbagnall1** [quarkusio/quarkus#32607] [Q&A] [answered]
Hi, I am currently converting an existing Application to use Quarkus. When I build the application I get the erorr 'Configuration methods cannot accept parameters' from io.smallrye.config.ConfigurationMappingInterface. The methods it is complaining about are when the configuration class either

a) has a method which accepts parameters,
or
b) has a method which returns an object which has methods which accept parameters.

Is it possible to override this check?

**Comment by radcortez** [comment, marked as answer]
No, because we generate the implementation of such methods, and we wouldn't know what to do with the method parameters.

You can add default methods to your mapping classes and add parameters or return any type you want.

Can you please post an example of your mappings?
